src/function.rs

```diff
@@ -288,6 +288,28 @@ where
         self.maybe_changed_after(db, input, revision, cycle_heads)
     }
 
+    fn minimum_serialized_edges(&self, zalsa: &Zalsa, edge: QueryEdge) -> Vec<QueryEdge> {
+        let input = edge.key().key_index();
+
+        let Some(memo) =
+            self.get_memo_from_table_for(zalsa, input, self.memo_ingredient_index(zalsa, input))
+        else {
+            return Vec::new();
+        };
+
+        let origin = memo.revisions.origin.as_ref();
+
+        // Collect the minimum dependency tree.
+        origin
```

Comment:
Shouldn't this include `edge` if the ingredient is serializable?

Reply:
I see. Should this function return a set? I assume that many queries will have overlapping dependencies or is this done a level higher up?

Reply:
That was being done a level up, but now it collects directly into the set.